packages/integration-tests/src/entities/UserCodegen.ts

```diff
@@ -163,92 +166,94 @@ export abstract class UserCodegen extends BaseEntity<EntityManager> {
     "likedByUsers",
     "comment_id",
   );
-
   constructor(em: EntityManager, opts: UserOpts) {
     super(em, userMeta, UserCodegen.defaultValues, opts);
-    setOpts(this as any as User, opts, { calledFromConstructor: true });
+    setOpts((this as any) as User, opts, { calledFromConstructor: true });
   }
-
   get id(): UserId | undefined {
     return this.idTagged;
   }
-
   get idOrFail(): UserId {
     return this.id || fail("User has no id yet");
   }
-
   get idTagged(): UserId | undefined {
     return this.__orm.data["id"];
   }
-
   get idTaggedOrFail(): UserId {
     return this.idTagged || fail("User has no id tagged yet");
   }
-
   get name(): string {
     return this.__orm.data["name"];
   }
-
   set name(name: string) {
     setField(this, "name", cleanStringValue(name));
   }
-
+  /**
+   * Case is preserved. If searching use an ILIKE.
+   */
   get email(): string {
     return this.__orm.data["email"];
   }
-
+  /**
+   * Case is preserved. If searching use an ILIKE.
+   */
   set email(email: string) {
     setField(this, "email", cleanStringValue(email));
   }
-
+  /**
+   * This is an example of a multi
+   * line comment with
+   *
+   * line breaks and also to show that we just use headings, and not the level, allowing for more customisation in users docs.
+   */
   get ipAddress(): IpAddress | undefined {
     return this.__orm.data["ipAddress"];
   }
-
+  /**
+   * This is an example of a multi
+   * line comment with
+   *
+   * line breaks and also to show that we just use headings, and not the level, allowing for more customisation in users docs.
+   */
```

Comment:
Do you know if tsdoc wants/expects comments on both the getter and the setter? I would have guessed when like viewing this in vscode, there is only one logical `ipAddress` property, so it would just like pick one, say the getter, and treat that as where comments should go.

Or maybe it like merges the getter+setter comments? Not sure...

Reply:
Yeah, I did a bit of a deep dive looking in to this before starting. I believe this is a bit messy at the moment:

- set & get are treated independently at a TSDoc level
- They are merged at usage. There's an open bug for this though (https://github.com/microsoft/TypeScript/issues/48801), so I didn't want to model how this works around what is marked as a bug. With the `fieldSourceType` property, it would be possible for CommentStores to tweak this, by deciding to only provide comments for getters.

Reply:
Hm, does the "merged at usage" apply to tsdoc output (not just internally)?

Kinda wondering if we could just annotate the getter and assume readers are either just looking at the getter, or using vscode / something that merges the getter+setter since typically the getter+setter is exposed as a single logical property.

Does that seem reasonable, or does the doc output you're using really look best with the comment included in both the getter + setter?

Reply:
I'm happy to move to just getters, it's an easy thing to change if there is ever movement on the above issue (which I think essentially refers to how the TS LSP exposes this stuff) and as it's touching code-generated files.

You can see the behavior here, hoving over the usages at the bottom. https://www.typescriptlang.org/play?#code/MYGwhgzhAEAKYCcCmA7ALtA3gKGtX0A9AFTEHTHQCS0A5kmmgJYq3RgC270KnSAdOWKEC9DLw5IAFAEos2AgF8FePARJlVlGhAbNW7LmB59BWkXl3i+s+UpWr1pIdUPc0ACyTQIpoRboGaCYAE1RmAE9bHHtyfDwNFxpOdy8fP3MCK2Cw9CYouRi8ZWVsFCQAdzhEcNl+CSQFcqr4ZHQ60PD8oA

Reply:
Oh, I see, thanks for the link! I thought "merged" was like "appended", but if they're the same, it's deduped, so then, yeah, seems like not a big deal to put them on either getters-only or setters-only or what not... Probably fine to leave as-is if/until we have a reason to do something different.